app/controllers/notes_controller.rb

```diff
@@ -20,6 +20,7 @@ def edit; end
   # POST /notes
   def create
     @note = Note.new(note_params)
+    @note.creator_user_id = current_user.id
```

Comment:
When I see it correctly we have a `has many` association between `User` and `Notes` (https://guides.rubyonrails.org/association_basics.html#the-has-many-association). Normally we would create a new `Note` for a `User` with `@note = User.notes.create(note_params)` (https://guides.rubyonrails.org/association_basics.html#the-has-many-association). Why are we not doing it here? 

We could achieve it with nested Ressources (https://guides.rubyonrails.org/routing.html#nested-resources).

Reply:
Thanks for the pointer to nested resources! I haven't met this concept so far so I do not know whether you should use it as often as possible or whether it is just another optional possibility. From a logical perspective, it might make sense to nest the notes after the users, but on the other hand, every user should only be able to see his own notes, so the extra route `/users/<user_id_that_must_be_the_current_user_id_and_nothing_else>/notes` feels kind of redundant. Do you see my point? Do you know more about the conventions for this concept?

Reply:
`@note.creator = current_user` should be supported, no?

Or you might also consider adding the current_user of a request to `note_params` if this is something that only logged in users (where `current_user` is set by [devise ](https://github.com/heartcombo/devise#controller-filters-and-helpers)) can do.

Reply:
I don't see an issue with that one